examples/offline_inference/rlhf_colocate.py

```diff
@@ -86,20 +90,45 @@ def __init__(self):
         from vllm.platforms import current_platform
 
         self.device_uuid = current_platform.get_device_uuid(0)
+        self.zmq_context = zmq.Context()
+        self.zmq_handle = f"ipc:///tmp/rl-colocate-zmq-{self.device_uuid}.sock"
 
     def report_device_id(self) -> str:
         return self.device_uuid
 
-    def get_weight_ipc_handles(self):
-        from torch.multiprocessing.reductions import reduce_tensor
+    def get_zmq_handles(self):
+        return {self.device_uuid: self.zmq_handle}
 
-        data = {}
-        for name, p in self.model.named_parameters():
-            # A training actor might hold only a subset of the weights and may
-            # need to gather weights from other actors. For demonstration
-            # purposes, each training actor owns the full weight set.
-            data[name] = reduce_tensor(p.detach())
-        return {self.device_uuid: data}
+    def update_weights(self):
+        s = self.zmq_context.socket(zmq.REQ)
+        s.bind(self.zmq_handle)
+        buffer = torch.empty(1 << 30, dtype=torch.uint8, device="cuda:0")
```

Comment:
in the training side, they can iterate the weights to get max size during init time, and pass the information to worker when setting up the zmq connection.